spring-cloud-modules/spring-cloud-open-telemetry/spring-cloud-open-telemetry1/src/test/java/com/baeldung/opentelemetry/controller/ProductControllerUnitTest.java

```diff
@@ -0,0 +1,92 @@
+package com.baeldung.opentelemetry.controller;
+
+import com.baeldung.opentelemetry.api.client.PriceClient;
+import com.baeldung.opentelemetry.exception.ProductNotFoundException;
+import com.baeldung.opentelemetry.model.Price;
+import com.baeldung.opentelemetry.model.Product;
+import com.baeldung.opentelemetry.repository.ProductRepository;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.HttpStatus;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.web.client.HttpServerErrorException;
+
+
+import static org.mockito.Mockito.when;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+
+@ExtendWith(SpringExtension.class)
+@WebMvcTest(ProductController.class)
+public class ProductControllerUnitTest {
```

Comment:
Let's remove all the `public`s here too.

Reply:
Done here.